extensions/sensors/GetMovementSensors.cpp

```diff
@@ -0,0 +1,109 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <memory>
+#include <algorithm>
+#include <cctype>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "utils/ByteArrayCallback.h"
+#include "core/FlowFile.h"
+#include "core/logging/Logger.h"
+#include "core/ProcessContext.h"
+#include "core/Relationship.h"
+#include "GetMovementSensors.h"
+#include "io/DataStream.h"
+#include "io/StreamFactory.h"
+#include "ResourceClaim.h"
+#include "utils/StringUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+std::shared_ptr<utils::IdGenerator> GetMovementSensors::id_generator_ = utils::IdGenerator::getIdGenerator();
+
+const char *GetMovementSensors::ProcessorName = "EnvironmentalSensors";
+
+core::Relationship GetMovementSensors::Success("success", "All files are routed to success");
+
+void GetMovementSensors::initialize() {
+  logger_->log_trace("Initializing EnvironmentalSensors");
+  // Set the supported properties
+  std::set<core::Property> properties;
+
+  setSupportedProperties(properties);
+  // Set the supported relationships
+  std::set<core::Relationship> relationships;
+  relationships.insert(Success);
+  setSupportedRelationships(relationships);
+}
+
+void GetMovementSensors::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) {
+  imu = RTIMU::createIMU(&settings);
+  if (imu) {
+    imu->IMUInit();
+    imu->setGyroEnable(true);
+    imu->setAccelEnable(true);
+  } else {
+    throw std::runtime_error("RTIMU could not be initialized");
+  }
+
+}
+
+GetMovementSensors::~GetMovementSensors() {
+}
+
+void GetMovementSensors::onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session) {
+  auto flow_file_ = session->create();
+
+  flow_file_->setSize(0);
+
+  if (imu->IMURead()) {
+    RTIMU_DATA imuData = imu->getIMUData();
+
+    if (imuData.accelValid) {
+      auto vector = imuData.accel;
+      std::string degrees = RTMath::displayDegrees("acceleration", vector);
+      flow_file_->addAttribute("ACCELERATION", degrees);
+    }
+    if (imuData.gyroValid) {
+      auto vector = imuData.gyro;
+      std::string degrees = RTMath::displayDegrees("gyro", vector);
+      flow_file_->addAttribute("GYRO", degrees);
+    }
+
```

Comment:
Same type of comment w/r/t error or exception cases. Didn't notice anything wrong but I imagine there could be failures on sensor polls that we'd want to at least log a warn or error for.

Reply:
Afraid there is very little from this library that we can translate, aside from a valid value being collected.